apps/jupyterhub/envs/dev/values.yaml

```diff
@@ -8,10 +8,39 @@ hub:
           - port: 6443
           - port: 443
   config:
+    Authenticator:
+      enable_auth_state: true
+    # Lookup URLs from
+    # https://<keycloak-url>/realms/<realm>/.well-known/openid-configuration
+    GenericOAuthenticator:
+      client_id: jupyterhub
+      client_secret: jupyterhub-client-secret
+      oauth_callback_url: https://jupyter.dev.k8tre.internal/hub/oauth_callback
+      authorize_url: https://keycloak.dev.k8tre.internal/realms/master/protocol/openid-connect/auth
+      token_url: https://keycloak.dev.k8tre.internal/realms/master/protocol/openid-connect/token
+      userdata_url: https://keycloak.dev.k8tre.internal/realms/master/protocol/openid-connect/userinfo
```

Comment:
A potential issue here is these are "external" hosts to the cluster. This is required for anything the browser needs to access, but it _should_ be possible for internal hub→keycloak requests to remain within the cluster. I'll see if using the internal cluster service domain (something like `keycloak.keycloak.svc.cluster.local`?) works.